src/main/kotlin/me/zeroeightsix/kami/util/MessageDetectionHelper.kt

```diff
@@ -0,0 +1,61 @@
+package me.zeroeightsix.kami.util
+
+import java.util.regex.Pattern
+
+object MessageDetectionHelper {
+    fun String.detectAndRemove(regex: Regexes): String? {
+        if (!this.detect(regex)) return null
+
+        val removed = regex.regex.replace(this, "")
+        return if (removed.isEmpty()) null else {
+            removed
+        }
+    }
+
+    fun String.detect(vararg regexes: Regexes): Boolean {
+        regexes.forEach {
+            if (it.regex.containsMatchIn(this)) return true
+        }
+        return false
+    }
+
+    fun String.detect(setting: Boolean, vararg regex: Regexes) = setting && this.detect(*regex)
+
+    fun isDirectReceived(direct: Boolean, message: String) =
+        message.detect(direct, Regexes.DIRECT, Regexes.DIRECT_ALT_1, Regexes.DIRECT_ALT_2)
+
+    fun isDirect(direct: Boolean, message: String) =
```

Comment:
i dont understand the purpose of the `direct` parameter. couldnt this just be checked outside if this function to remove some really specific code thatll probably never be used otherwise?

Reply:
This is legacy from KAMI Blue, will remove.